src/repr/scalar/mod.rs

```diff
@@ -764,6 +764,26 @@ impl Default for Interval {
     }
 }
 
+impl PartialEq for Interval {
+    fn eq(&self, other: &Self) -> bool {
+        let dur_match = if self.duration.as_nanos() == other.duration.as_nanos() {
+            self.duration.as_nanos() == 0 || (self.is_positive_dur == other.is_positive_dur)
+        } else {
+            false
+        };
+
+        dur_match && (self.months == other.months)
+    }
+}
+
+impl Hash for Interval {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.months.hash(state);
+        self.duration.hash(state);
+        self.is_positive_dur.hash(state);
```

Comment:
Am I crazy, or is this exactly the same as the `Hash` implementation you'd get if you derived it automatically? I think this wants to take the new `PartialEq` behavior into account?

Reply:
smh; you are correct